### PR TITLE
sys-fs/lvm2: switch to positive USE=lvm logic

### PR DESCRIPTION
**Pull request by SoapGentoo**
Closes: https://bugs.gentoo.org/718910
Signed-off-by: David Seifert <soap@gentoo.org>

### CODE REVIEW

File: sys-fs/lvm2/lvm2-2.03.14-r4.ebuild [hunk 1]
```diff
@@ -13,8 +14,8 @@ SRC_URI="ftp://sourceware.org/pub/lvm2/${PN/lvm/LVM}.${PV}.tgz
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~alpha amd64 arm arm64 hppa ~ia64 ~loong ~m68k ~mips ppc ppc64 ~riscv ~s390 sparc x86 ~amd64-linux ~x86-linux"
-IUSE="readline static static-libs systemd lvm2create-initrd sanlock selinux +udev +thin device-mapper-only"
-REQUIRED_USE="device-mapper-only? ( !lvm2create-initrd !sanlock !thin )
+IUSE="readline static static-libs systemd lvm lvm2create-initrd sanlock selinux +udev +thin"
```
Comment (thesamesam) [inline sys-fs/lvm2/lvm2-2.03.14-r4.ebuild:17]: I feel like we need a news item for this if we're not going to default it on (and I don't want to default it on)

File: sys-fs/lvm2/lvm2-2.03.14-r4.ebuild [hunk 2]
```diff
@@ -34,7 +35,7 @@ RDEPEND="${DEPEND_COMMON}
 	>=sys-apps/baselayout-2.2
```
Comment (thesamesam) [inline sys-fs/lvm2/lvm2-2.03.14-r4.ebuild:35]: If you didn't already, can you audit the deps to make sure they're minimal for USE=-lvm?

File: sys-fs/lvm2/lvm2-2.03.16-r1.ebuild [hunk 3]
```diff
@@ -190,16 +190,14 @@ src_configure() {
 	)
 	# Hard-wire this to bash as some shells (dash) don't know
 	# "-o pipefail" #682404
-	CONFIG_SHELL="/bin/bash" \
+	CONFIG_SHELL="${BROOT}/bin/bash" \
```
Comment (thesamesam) [inline sys-fs/lvm2/lvm2-2.03.16-r1.ebuild:193]: ❤️

**Review by thesamesam — COMMENTED** [review]
I like it!

File: sys-fs/lvm2/lvm2-2.03.14-r4.ebuild (hunk 1 above)
Comment (SoapGentoo) [reply sys-fs/lvm2/lvm2-2.03.14-r4.ebuild:17]: better idea: let's `+lvm` for now, and then prepare the news item. Doing news item now means this PR has to wait another X weeks before we can actually change the negative logic to positive

File: app-emulation/libvirt/libvirt-7.7.0-r3.ebuild [hunk 4]
```diff
@@ -84,7 +84,7 @@ RDEPEND="
 	iscsi? ( sys-block/open-iscsi )
 	iscsi-direct? ( >=net-libs/libiscsi-1.18.0 )
 	libssh? ( net-libs/libssh )
-	lvm? ( >=sys-fs/lvm2-2.02.48-r2[-device-mapper-only(-)] )
+	lvm? ( >=sys-fs/lvm2-2.02.48-r2[lvm] )
```
Comment (thesamesam) [inline app-emulation/libvirt/libvirt-7.7.0-r3.ebuild:87]: These all need (+) right? Ugh, but the interaction with device-mapper-only...

Comment (thesamesam) [reply app-emulation/libvirt/libvirt-7.7.0-r3.ebuild:87]: oh nvm I see you did all versions, ok

File: sys-fs/lvm2/lvm2-2.03.16-r1.ebuild [hunk 5]
```diff
@@ -190,22 +195,19 @@ src_configure() {
...
 }
 
 src_compile() {
-	pushd include >/dev/null
-	emake V=1
-	popd >/dev/null
+	emake -C include V=1
 
-	if use device-mapper-only ; then
+	if ! use lvm ; then
```
Comment (thesamesam) [inline sys-fs/lvm2/lvm2-2.03.16-r1.ebuild:204]: Flip this around?

File: sys-fs/lvm2/lvm2-2.03.16-r1.ebuild [hunk 6]
```diff
@@ -219,34 +221,40 @@ src_install() {
 	local INSTALL_TARGETS=( install install_tmpfiles_configuration )
 	# install systemd related files only when requested, bug #522430
 	use systemd && INSTALL_TARGETS+=( SYSTEMD_GENERATOR_DIR="$(systemd_get_systemgeneratordir)" install_systemd_units install_systemd_generators )
```
Comment (thesamesam) [inline sys-fs/lvm2/lvm2-2.03.16-r1.ebuild:223]: I wonder if it'd (all these INSTALL_TARGET appends) look better in a big array with usev/echo but let's not worry for now

**Review by thesamesam — APPROVED** [review]

File: sys-fs/lvm2/lvm2-2.03.16-r1.ebuild [hunk 7]
```diff
@@ -215,38 +217,48 @@ src_test() {
...
-	use device-mapper-only && INSTALL_TARGETS=( install_device-mapper )
+	local INSTALL_TARGETS=(
+		# full LVM2
+		$(usev lvm "install install_tmpfiles_configuration")
+		# install systemd related files only when requested, bug #522430
+		$(usev $(usex lvm systemd lvm) "SYSTEMD_GENERATOR_DIR=$(systemd_get_systemgeneratordir) install_systemd_units install_systemd_generators")
+
+		# install dm unconditionally
+		install_device-mapper
+	)
```
Comment (SoapGentoo) [inline sys-fs/lvm2/lvm2-2.03.16-r1.ebuild:228]: I've checked that the install trees are in fact identical

File: sys-fs/lvm2/lvm2-2.03.16-r1.ebuild (hunk 6 above)
Comment (SoapGentoo) [reply sys-fs/lvm2/lvm2-2.03.16-r1.ebuild]: dunn

**Review by floppym — APPROVED** [review]

**Pull request closed without merging**